### ISSUE TITLE
Connect go generated code looks will get wrong import path

### ISSUE THREAD

**Issue by renyijiu** [bufbuild/connect-go#412]
**Describe the bug**

We have some additional proto definitions to put here [workflow](https://buf.build/coscene-io/argo-workflows-apis), and another repo will reference some of them, roughly as follows

```proto
syntax = "proto3";

package coscene.dataplatform.v1alpha1.services;

import "coscene/dataplatform/v1alpha1/resources/workflow.proto";
import "coscene/workflow/v1alpha1/workflow.proto";
import "google/api/annotations.proto";
import "google/api/field_behavior.proto";
import "google/api/resource.proto";
import "google/protobuf/field_mask.proto";

option go_package = "github.com/coscene-io/cosceneapis/dataplatform/genproto/v1alpha1/services;servicesv1alpha1";

// Service with interacting workflow templates.
service WorkflowTemplateService {
  // Retrieve a workflow template.
  rpc GetWorkflowTemplate(GetWorkflowTemplateRequest) returns (coscene.workflow.v1alpha1.WorkflowTemplate) {
    option (google.api.http) = {
      get: "/dataplatform/v1alpha1/{name=workflowTemplates/*}"
    };
  }
}
```

Then the generated connect-go package got an import error
<img width="1353" alt="image" src="https://user-images.githubusercontent.com/8318266/205848850-63876602-5173-4bcb-a30b-1a9fe8551cff.png">

it looks should import `v1alpha1 "buf.build/gen/go/coscene-io/argo-workflows-apis/protocolbuffers/go/coscene/workflow/v1alpha1"`,
but got import `v1alpha1 "buf.build/gen/go/coscene-io/argo-workflows-apis/bufbuild/connect-go/coscene/workflow/v1alpha1"`, then got import error.

**Environment (please complete the following information):**
- `connect-go` version or commit: v1.3.1-20221112072124-2695da33f313.1
- `go version`:  go1.19 darwin/arm64

- your complete `go.mod`:

```go
module github.com/coscene-io/cos

go 1.19

require (
	buf.build/gen/go/coscene-io/cosceneapis/bufbuild/connect-go v1.3.1-20221202034448-6407b280bf92.1
	buf.build/gen/go/coscene-io/cosceneapis/protocolbuffers/go v1.28.1-20221202034448-6407b280bf92.4
	github.com/ghodss/yaml v1.0.0
	github.com/knadh/koanf v1.4.4
	github.com/minio/minio-go/v7 v7.0.44
	github.com/oriser/regroup v0.0.0-20210730155327-fca8d7531263
	github.com/pkg/errors v0.9.1
	github.com/sirupsen/logrus v1.9.0
	github.com/spf13/cobra v1.6.1
	golang.org/x/sync v0.1.0
	google.golang.org/protobuf v1.28.1
)

require (
	buf.build/gen/go/coscene-io/argo-workflows-apis/bufbuild/connect-go v1.3.1-20221112072124-2695da33f313.1 // indirect
	buf.build/gen/go/coscene-io/argo-workflows-apis/protocolbuffers/go v1.28.1-20221112072124-2695da33f313.4 // indirect
	github.com/coscene-io/x v0.0.0-20221111073514-195ce36aca02 // indirect
	github.com/bufbuild/connect-go v1.3.1
	github.com/dustin/go-humanize v1.0.0 // indirect
	github.com/fsnotify/fsnotify v1.6.0 // indirect
	github.com/golang/protobuf v1.5.2 // indirect
	github.com/google/uuid v1.3.0 // indirect
	github.com/inconshreveable/mousetrap v1.0.1 // indirect
	github.com/json-iterator/go v1.1.12 // indirect
	github.com/klauspost/compress v1.15.9 // indirect
	github.com/klauspost/cpuid/v2 v2.1.0 // indirect
	github.com/minio/md5-simd v1.1.2 // indirect
	github.com/minio/sha256-simd v1.0.0 // indirect
	github.com/mitchellh/copystructure v1.2.0 // indirect
	github.com/mitchellh/mapstructure v1.5.0 // indirect
	github.com/mitchellh/reflectwalk v1.0.2 // indirect
	github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect
	github.com/modern-go/reflect2 v1.0.2 // indirect
	github.com/rs/xid v1.4.0 // indirect
	github.com/spf13/pflag v1.0.5 // indirect
	golang.org/x/crypto v0.0.0-20220722155217-630584e8d5aa // indirect
	golang.org/x/net v0.0.0-20220926192436-02166a98028e
	golang.org/x/sys v0.2.0 // indirect
	golang.org/x/text v0.4.0 // indirect
	google.golang.org/genproto v0.0.0-20220926220553-6981cbe3cfce // indirect
	gopkg.in/ini.v1 v1.66.6 // indirect
	gopkg.in/natefinch/lumberjack.v2 v2.0.0
	gopkg.in/yaml.v2 v2.4.0 // indirect
)

```

**Additional context**

**Comment by renyijiu** [comment]
> I haven't been able to recreate this; would you be able to supply your `buf.yaml` and `buf.gen.yaml` too?

I have built a simple example project with the same error problem, https://github.com/renyijiu/connect-playground.

I'm not sure if the proto file corresponding to the [workflow](https://buf.build/angry-popcorn/argo-workflows-apis/tree/main) does not conform to the buf specification, which causes this problem

**Comment by srikrsna** [comment]
Hey @renyijiu! Thank you for the detailed example. I was just about to reply. We've identified the issue and the fix is live.
This issue was with Remote Packages.

In the example, run
```
go get buf.build/gen/go/angry-popcorn/connect-playground/bufbuild/connect-go@v1.3.1-20221208033112-ab1a824519f0.1
```
This will fix it. 

Unfortunately, it won't work for used versions of remote packages. But newer versions will definitely work. Older versions that have not been depended on will also work. That is exactly how I tested the example you provided using a slightly older plugin version.

**Comment by srikrsna-buf** [comment]
Hey @renyijiu! Thank you for the detailed example. I was just about to reply. We've identified the issue and the fix is live.
This issue was with Remote Packages.

In the example, run
```
go get buf.build/gen/go/angry-popcorn/connect-playground/bufbuild/connect-go@v1.3.1-20221208033112-ab1a824519f0.1
```
This will fix it. 

Unfortunately, it won't work for used versions of remote packages. But newer versions will definitely work. Older versions that have not been depended on will also work. That is exactly how I tested the example you provided using a slightly older plugin version.
